packages/fuels-contract/src/contract.rs

```diff
@@ -512,16 +513,40 @@ where
         .await
     }
 
+    /// Simulates calls and attempts to set output variables based on the received error.
+    /// Forwards the reiceived error if it cannot be fixed.
+    pub async fn try_resolve(mut self, max_attempts: Option<u64>) -> Self {
```

Comment:
I feel like the naming should be improved. `try_resolve` is too broad, and looks like a Rust method. Something like `auto_set_variable_outputs` would be better IMO, what do you think ?

Reply:
`try_resolve` is just a placeholder but we will need something more general than `auto_set_variable_outputs ` since this method will also set message outputs and external contracts.